controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+
+	if homeClusterName == "" {
+		return errorswrapper.New("failed to find new home cluster")
+	}
+
+	err := r.deleteExistingManfiestWork(avr, subscription)
+	if err != nil {
+		return err
+	}
+
+	err = r.restoreBackedupPVs(subscription, homeClusterName)
+	if err != nil {
+		return err
+	}
+
+	return r.unpauseSubscription(subscription)
+}
+
+func (r *ApplicationVolumeReplicationReconciler) findVRGManifestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) (bool, error) {
+	const notFound = false
+
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrgMWName, Namespace: d.HomeCluster}, mw)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("ManifestWork not found for Subscription", "name", subscription.Name)
+
+				return notFound, nil
+			}
+
+			return notFound, errorswrapper.Wrap(err, "failed to retrieve manifestwork")
+		}
+
+		return found, nil
+	}
+
+	return notFound, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) deleteExistingManfiestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrgMWName, Namespace: d.HomeCluster}, mw)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+
+			r.Log.Error(err, "Could not fetch ManifestWork")
+
+			return errorswrapper.Wrap(err, "failed to retrieve manifestWork")
+		}
+
+		r.Log.Info("deleting ManifestWork", "name", mw.Name)
+
+		return r.Client.Delete(context.TODO(), mw)
 	}
 
 	return nil
 }
 
-func (r *ApplicationVolumeReplicationReconciler) createVRGManifestWork(
-	subscription subv1.Subscription) (string, string, error) {
-	const empty string = ""
+func (r *ApplicationVolumeReplicationReconciler) restoreBackedupPVs(
+	subscription *subv1.Subscription, homeClusterName string) error {
+	r.Log.Info("Restoring PVs to managed cluster", "name", homeClusterName)
 
-	// Select cluster decisions from each Subscriptions of an app.
-	homeCluster, peerCluster, err := r.selectPlacementDecision(&subscription)
+	// TODO: get PVs from S3
+	pvList, err := r.listPVsFromS3Store()
 	if err != nil {
-		return empty, empty, fmt.Errorf("failed to get placement targets from subscription %s with error: %w",
-			subscription.Name, err)
+		return errorswrapper.Wrap(err, "failed to retrieve PVs from S3 store")
 	}
 
-	if homeCluster == "" {
-		return empty, empty,
-			fmt.Errorf("no home cluster configured in subscription %s", subscription.Name)
+	r.Log.Info("PV k8s objects", "len", len(pvList))
```

Comment:
(nit/future with S3 integration) There maybe no PVs (or no PVs backed up yet due to async processing of VRG on MC and races due to the same)

Reply:
Right, we should just return